bin/reorder-keys

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+if [ ! -d exercises ] ; then
+    echo "Error: \"exercises\" directory not found, make sure you're running from the repository root" >&2
+    exit 1
+fi
+
+tmp_file='.reordered-canonical-data.json'
+
+for canonical_data_file in exercises/*/canonical-data.json ; do
+    jq 'walk(if (type == "object") and has("uuid") then . as $obj | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key ({}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end)' "$canonical_data_file" > "$tmp_file" && mv "$tmp_file" "$canonical_data_file"
+done
```

Comment:
As a follow up on the line splitting: maybe something like the below. But it's fine. PR it if you want.

The default shell for GitHub Actions on non-Windows is bash with `-e`, so I have a slight preference for adding that at the top of the script to get consistent behavior locally. Then we can put the `mv` on a separate line without the `&&`.

```suggestion
#!/usr/bin/env bash

set -e

if [ ! -d exercises ] ; then
    echo "Error: \"exercises\" directory not found, make sure you're running from the repository root" >&2
    exit 1
fi

tmp_file='.reordered-canonical-data.json'

reorder() {
    jq 'walk(if (type == "object") and has("uuid") then . as $obj
        | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key(
            {}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end
        )' "$1" > "$tmp_file"
    mv "$tmp_file" "$1"
}

for canonical_data_file in exercises/*/canonical-data.json ; do
    reorder "$canonical_data_file"
done
```